docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,85 @@
+  - Both application-services team and the application teams generally put low priority on updating the application-services version.  We generally focus on the benefits of waiting to take releases rather than the costs.
+
+Mozilla generally favors a rapid release model to long cycle times.  Firefox desktop adopted rapid releases over a decade ago and there is current work to move both the mobile applications and our services towards rapid releases.  Although there are some costs to rapid releases, as a company we've decided that we're willing to accept them.
+
+Application services should consider moving towards a rapid release model.  The same reasoning that led other teams to adopt rapid releases also applies to us.  Furthermore, the mobile teams are moving towards rapid releases which may depend on us also adopting them.  Finally, we are in the process of [moving towards a single monorepo for android code](https://docs.google.com/document/d/1VMr5rtn-2Ilb4DSjNoD6D9bk-RX1GGOin8Hpn9sqMbA/edit), and it's somewhat likely that the monorepo will contain the iOS code as will.  A monorepo would require that application services code is taken by consumer applications as soon as it's merged.  Reducing the amount of time between our releases and when applications take those releases brings us closer to that monorepo world.
+
+## Decision Drivers
+
+* We want to get changes merged and released to our consumer applications faster.
+* We want to simplify pushing new code to applications, especially urgent bug fixes.
+* We want to move towards the future monorepo world.
+* Mozilla is moving towards rapid releases for mobile code.
+* We are willing to accept the overhead of upgrading application-services more frequently in application repositories.
+
+## Considered Options
+
+* **(A) Keep our current release process**
+    * Keep everything the same for a while and wait for other code to move into the monorepo.
+* **(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+    * Phase 1: update our branching model and release process
+        * Update our branches to match how moz-central: we would have a `main`, `beta` and `release` branch, with nightly being built off of main, as well as `esr-*` branches for older ESR releases.
```

Comment:
Why not use version-specific branches like the mobile repos do? Would avoid the messy merges every cycle.

Reply:
To expand on this, it might be worth approaching this from a similar angle that NSS is currently using: regular updates between NSS tip and m-c during the Nightly cycle (basically in the form of regular A-S nightlies here), then RTM at the end of the Nightly cycle (i.e. creating a vXXX branch off main here), then point releases as needed during beta/release.

Reply:
This is also very close in concept to how pre-monorepo Android Components integrated in Fenix & Focus.

Reply:
@rvandermeulen (sorry, what is "RTM" in this context?)

Reply:
NSS tags the final release (RTM) at the end of the nightly cycle. See https://hg.mozilla.org/projects/nss/tags

Reply:
Thanks!  This seems much simpler to me as well.